os/fs/driver/mtd/smart.c

```diff
@@ -2232,24 +2210,10 @@ static int smart_scan(FAR struct smart_struct_s *dev)
 		}
 	}
 #endif
-
-#ifdef CONFIG_MTD_SMART_JOURNALING
-	if (dev->formatstatus == SMART_FMT_STAT_FORMATTED) {
-		ret = smart_journal_init(dev);
-		if (ret < 0) {
-			fdbg("journal init failed ret : %d\n", ret);
-			goto err_out;
-		}
-	}
-#endif
 	ret = OK;
 
 err_out:
-#ifdef CONFIG_MTD_SMART_JOURNALING
-	if (dev->sSeqLogMap != NULL) {
-		kmm_free(dev->sSeqLogMap);
```

Comment:
There is a memory allocation for `rootdirdev`. But there is no free for failure case. Is it okay?

Reply:
yes it should be kept. it is block driver's private data, all the time it should be kept until smartfs unmounted.